### PR TITLE
Using search string functionality

### PR DESCRIPTION
**Pull request by Will-Cooper**
Adding filter to main search page, plus additional page for loose string search

### CODE REVIEW

**Comment by dr-rodriguez** [comment]
Tried it out and I immediately see how much more useful the `Loose Database Query` is. This was a good call. I do think we can name it more descriptively, though since I'm not sure 'loose' really captures what it does. Full? General? Not sure. 

Thoughts on what we could call this search type, @kelle?

File: simple_app/utils.py [hunk 1]
```diff
@@ -111,7 +134,16 @@ class SearchForm(FlaskForm):
...
+    search = StringField('Search for an object:', id='mainsearchfield')  # searchbar
+    refsearch = StringField('Filter by reference or comment:', id='refsearchfield')
+    submit = SubmitField('Query', id='querybutton')  # clicker button to send request
+
+
+class LooseSearchForm(FlaskForm):
+    """
+    Searchbar class
+    """
+    search = StringField('Search for anything:', id='mainsearchfield')  # searchbar
```
Comment (dr-rodriguez) [inline simple_app/utils.py:146]: I'd be a bit more descriptive here. Maybe something like: `Search all columns in all tables for:`

File: simple_app/utils.py [hunk 2]
```diff
@@ -111,7 +134,16 @@ class SearchForm(FlaskForm):
     """
     Searchbar class
     """
-    search = StringField('Search for an object:', id='autocomplete')  # searchbar
+    search = StringField('Search for an object:', id='mainsearchfield')  # searchbar
+    refsearch = StringField('Filter by reference or comment:', id='refsearchfield')
```
Comment (dr-rodriguez) [inline simple_app/utils.py:138]: Do we need the Filter by reference or comment part if we also have the Loose search? The later does the same and is a little more powerful.

File: simple_app/app_simple.py [hunk 3]
```diff
@@ -28,28 +28,70 @@ def search():
...
+    """
+    form = LooseSearchForm()  # main searchbar
+    if (query := form.search.data) is None:  # content in main searchbar
+        query = ''
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    results: Dict[str, pd.DataFrame] = db.search_string(query, fmt='pandas', verbose=False)  # search
+    resultsout = {}
+    if len(results):
+        for tabname, df in results.items():
+            if tabname in REFERENCE_TABLES:  # don't want the reference tables
```
Comment (dr-rodriguez) [inline simple_app/app_simple.py:79]: I disagree; I think in this case we do want the reference tables

File: simple_app/utils.py (hunk 2 above)
Comment (Will-Cooper) [reply simple_app/utils.py]: as discussed

File: simple_app/app_simple.py (hunk 3 above)
Comment (Will-Cooper) [reply simple_app/app_simple.py]: sure

**Review by dr-rodriguez — APPROVED** [review]

**Pull request merged**